From: SeongJae Park <sj@kernel.org>
Date: Wed, 7 Jan 2026 17:44:14 -0800
Subject: [PATCH] ==== hide kdamond and kdamond_lock ====

mm/damon: hide kdamond and kdamond_lock from API callers

'kdamond' and 'kdamond_lock' fields are initially exposed to DAMON API
callers for flexible synchronization and use cases.  As DAMON API became
somewhat complicated compared to the early days, Keeping those exposed
could only encourage the API callers to invent more creative but
complicated and difficult-to-debug use cases.

Fortunately DAMON API callers didn't invent that many creative use
cases.  There exist only two use cases of 'kdamond' and 'kdamond_lock'.
Finding whether the kdamond is actively running, and getting the pid of
the kdamond.  For the first use case, a dedicated API function, namely
'damon_is_running()' is provided, and all DAMON API callers are using
the function for the use case.  Hence only the second use case is where
the fields are directly being used by DAMON API callers.

To prevent future invention of complicated use cases of the fields, hide
the fields from the API callers.  For that, provide a new dedicated
DAMON API functions for the remaining use case, namely
damon_kdamond_pid(), migrate DAMON API callers to use the new function,
and mark the fields as private fields.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_oaoqohca | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_oaoqohca

diff --git a/hkml_cv_bogus/hkml_cv_bogus_oaoqohca b/hkml_cv_bogus/hkml_cv_bogus_oaoqohca
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.47.3

